cinn/runtime/cuda/cuda_util.cc

```diff
@@ -283,72 +213,78 @@ void cinn_gpu_cublas_gemm(const std::vector<int> &attrs,
 
   const float *lhs_data  = reinterpret_cast<const float *>(lhs->memory);
   const float *rhs_data  = reinterpret_cast<const float *>(rhs->memory);
-  const float *bias_data = reinterpret_cast<const float *>(bias->memory);
+  const float *bias_data = bias ? reinterpret_cast<const float *>(bias->memory) : nullptr;
   float *output_data     = reinterpret_cast<float *>(output->memory);
 
-  CHECK_GE(attrs.size(), 12);
-  int lhs_dim_size  = attrs[attrs.size() - 6];
-  int rhs_dim_size  = attrs[attrs.size() - 5];
-  int bias_dim_size = attrs[attrs.size() - 4];
-  bool lhs_trans    = static_cast<bool>(attrs[attrs.size() - 3]);
-  bool rhs_trans    = static_cast<bool>(attrs[attrs.size() - 2]);
+  CHECK_GE(attrs.size(), 13);
+  int lhs_dim_size = attrs[attrs.size() - 7];
+  int rhs_dim_size = attrs[attrs.size() - 6];
+  int out_dim_size = attrs[attrs.size() - 5];
+  bool lhs_trans   = static_cast<bool>(attrs[attrs.size() - 4]);
+  bool rhs_trans   = static_cast<bool>(attrs[attrs.size() - 3]);
+  bool out_trans   = static_cast<bool>(attrs[attrs.size() - 2]);
+  // 1）C = A^T * B    -->  C^T = B^T * A
+  // 2）C = A * B^T    -->  C^T = B * A^T
+  // 3）C = A^T * B^T  -->  C^T = B * A
+  // 4）C = A * B      -->  C^T = B^T * A^T
+  if (out_trans) {
+    lhs_trans = static_cast<bool>(attrs[attrs.size() - 3]) ^ out_trans;
```

Comment:
不大明白这儿为啥要写这么复杂，直接
```suggestion
    lhs_trans = !lhs_trans;
```
不好么

Reply:
是不对？当然也可以这么写：
```cpp
auto tmp = lhs_trans;
lhs_trans = !rhs;
rhs = !tmp;
```
但是我觉得使用异或更好理解:`(A^T)^T=A`， `(A)^T=A^T`。